server/src/actors/events.rs

```diff
@@ -430,35 +444,50 @@ impl Handler<HandleEvent> for EventManager {
-                                        ProcessingError::RateLimited(secs)
+                        if let Some(store_forwarder) = store_forwarder {
+                            log::trace!("sending event to kafka {}", event_id);
+                            let request = StoreEvent {
+                                payload: processed,
+                                event_id,
+                            };
+                            let future = store_forwarder
+                                .send(request)
+                                .map_err(ProcessingError::ScheduleFailed)
+                                .and_then(move |result| {
+                                    result.map_err(move |error| ProcessingError::StoreFailed(error))
+                                });
+                            Box::new(future) as ResponseFuture<_, _>
+                        } else {
+                            log::trace!("sending event to sentry endpoint {}", event_id);
+                            let request = SendRequest::post(format!("/api/{}/store/", project_id))
+                                .build(move |builder| {
+                                    if let Some(origin) = meta.origin() {
+                                        builder.header("Origin", origin.to_string());
                                     }
-                                    other => ProcessingError::SendFailed(other),
+
+                                    builder
+                                        .header("X-Sentry-Auth", meta.auth().to_string())
+                                        .header("X-Forwarded-For", meta.forwarded_for())
+                                        .header("Content-Type", "application/json")
+                                        .body(processed)
+                                });
+                            let future = upstream
+                                .send(request)
+                                .map_err(ProcessingError::ScheduleFailed)
+                                .and_then(move |result| {
+                                    result.map_err(move |error| match error {
+                                        UpstreamRequestError::RateLimited(secs) => {
+                                            project.do_send(RetryAfter { secs });
+                                            ProcessingError::RateLimited(secs)
+                                        }
+                                        other => ProcessingError::SendFailed(other),
+                                    })
                                 })
-                            })
-                            .inspect(move |_| {
-                                metric!(timer("event.total_time") = start_time.elapsed())
-                            })
+                                .inspect(move |_| {
+                                    metric!(timer("event.total_time") = start_time.elapsed())
+                                });
```

Comment:
Missed this in my previous review: Please move the `inspect` call out of this closure onto the outer future (right before `into_actor`. That way we measure the total time in both branches.